pkl-doc/src/main/kotlin/org/pkl/doc/Util.kt

```diff
@@ -119,18 +118,11 @@ internal fun String.replaceSourceCodePlaceholders(
     .replace("%{endLine}", sourceLocation.endLine.toString())
 }
 
-/**
- * Encodes a URI string, encoding characters that are part of URI syntax.
- *
- * Follows `encodeURIComponent` from ECMAScript.
- */
+/** Encodes a URI string, encoding characters that are part of URI syntax. */
 internal val String.uriEncodedComponent
   get(): String {
-    val ret = URLEncoder.encode(this, StandardCharsets.UTF_8)
-    // Replace `+` with `%20` to be safe
-    // (see
-    // https://stackoverflow.com/questions/2678551/when-should-space-be-encoded-to-plus-or-20#:~:text=%20%20is%20a%20valid%20way,encodeURIComponent()%20does%20in%20JavaScript.)
-    return ret.replace("+", "%20")
+    val ret = URI(null, null, this, null)
+    return ret.toString().replace("/", "%2F")
```

Comment:
Why the change here?

Reply:
The previous implementation was encoding parentheses in percent encoding.